### ISSUE TITLE
Change remaining `maid` references

### ISSUE THREAD

**Issue by whit537** [capitalone/cloud-custodian#779]
I gather that the project, [initially](https://github.com/capitalone/cloud-custodian/commit/5aa46308651d913d285039477b9141cf6ef3965d) called `janitor`, has gone through a couple name changes (though I don't find issues or PRs discussing either):

- to `maid` in a9c508d974b356993eb70e9cb33c86276cecb719, and
- to `custodian` in 6e92a203d59524b3c3060b951e5cd109e6a94ca4.

I don't find any more references to `janitor` (only to Netflix's Janitor Monkey, by way of comparison). I do find a few remaining references to `maid`, however, and I'm interested in extirpating those, to alleviate future users and developers of unnecessary cognitive overhead ("Wait ... what the heck is Maid?"). I've picked off the least controversial in https://github.com/capitalone/cloud-custodian/pull/778. The rest appear to be more difficult to change, because they are in tag names that we depend on (e.g., [`maid_offhours`](https://github.com/capitalone/cloud-custodian/blob/94d51aa15822cf0c60b4881616ee29b5a536cdf1/c7n/filters/offhours.py#L174)) or even a [message format specifier](https://github.com/capitalone/cloud-custodian/blob/94d51aa15822cf0c60b4881616ee29b5a536cdf1/c7n/actions.py#L380). More research required!

**Comment by kapilt** [comment]
Yeah those name changes where pre-open sourcing of the code base and we had many extant internal users so places where there would have been impacted by the name change in the code, where left as is. Historical raisons

**Comment by whit537** [comment]
I think there's another PR or two of non-breaking changes to be had within the test suite before we have to start thinking more carefully about backcompat. Gonna see if I can discover those ...

**Comment by whit537** [comment]
More easy ones in https://github.com/capitalone/cloud-custodian/pull/780.

**Comment by whit537** [comment]
A more complicated one in #781.

**Comment by whit537** [comment]
Okay! Down to the real work here. I find five things to change:

1. [`maidmsg/1.0`](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/c7n/actions.py#L380)
1. [`CloudMaid`](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/c7n/output.py#L34)
1. [`maid_offhours`](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/c7n/filters/offhours.py#L174)
1. [`maid_status`](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/c7n/tags.py#L32)
1. `maid-*` in [`create_encrypted_volume`](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/c7n/resources/ebs.py#L664-L713)

**Comment by whit537** [comment]
For (5) it looks like we don't read that value anywhere, we just write it as an aid to disaster recovery.

**Comment by whit537** [comment]
For (3) and (4) we should be able to provide backcompat easily enough by allowing `DEFAULT_TAG` to be overriden somehow if we don't already (config file? cf. #401). We'd want a policy for migrating old tags to new for (3) (4) and (5).

Note that for (5) there's an asymmetry in the last tag compared to the others (`crypto` vs `crypt`). I suspect that's unintentional and could be cleaned up while we're here.

**Comment by whit537** [comment]
Similarly with (2), we should be able to expose the namespace to configuration if we don't already, change the default, and provide a policy for migrating (if that is sensical/possible in this case).

**Comment by whit537** [comment]
That leaves (1), which is the one I'm least confident about. What SQS consumers depend on that value? If it's just our own [mail worker](https://github.com/capitalone/cloud-custodian/blob/c4656fc0e495363a402b1cd3512122d1db0f8857/tools/c7n_mailer/c7n_mailer/worker.py#L23) then that's one thing: we can read both old and new values there if need be. If we consider that public API then the situation is more complicated.

**Comment by whit537** [comment]
@kapilt Do we need to go through a full deprecation/migration cycle for all five of those? Any jump out at you as "Oh, that's just internal we can cut over without coordinating with users"?

**Comment by kapilt** [comment]
1) should be okay to switch, only extant consumer is mail worker afaik.
2) we have historical data, as do any extant users under this metrics namespace
3-5) ditto re historical data except the store is the resources extant.

migration would need to be explicit and work for extant users doing an upgrade, i think we need to work out the deprecation/migration cycle for policy syntax, there's a bit i'd like to get cleaned up rather then continually adding compatibility, with some notion of having deprecation warnings on policy statements with and a quarterly rotation tied to semantic versioning.

one sans migration option would be to make these fields required, so users that have the implicit defaults would have to specify the tag. this would create a validation error on upgrade. this would require release notes documentation and communications out to the mailing list and gitter channels. th

**Comment by jantman** [comment]
As a stopgap, would it be possible to add a comment to the Monitoring section of the docs about the [CloudMaid namespace for CloudWatch metrics](https://github.com/capitalone/cloud-custodian/blob/8b812f605b246ed096cdcb4ee733b5162c2946f8/c7n/output.py#L34)? I assumed CW metrics were broken, since CW Logs go to ``cloud-custodian``, until I found the reference to the CloudMaid namespace in the source.

**Comment by kapilt** [comment]
yeah that would be useful, its actually possible to customize the target namespace now as well on cli
